src/ejabberd_c2s.erl

```diff
@@ -1759,6 +1768,11 @@ handle_info({broadcast, Type, From, Packet}, StateName, StateData) ->
     fsm_next_state(StateName, StateData);
 handle_info(dont_ask_offline, StateName, StateData) ->
     fsm_next_state(StateName, StateData#state{ask_offline = false});
+handle_info(close, StateName, StateData) ->
+    ?DEBUG("Timeout waiting for stream management acknowledgement of ~s",
+	   [jid:to_string(StateData#state.jid)]),
+    close(self()),
+    fsm_next_state(StateName, StateData);
```

Comment:
I don't think that's necessary, as the connection that resumes the session has a new fresh `#state` and `mgmt_ack_timer` is not [inherited from the old state][1].  But we could of course set `mgmt_ack_timer: undefined` to clarify the code.

[1]: https://github.com/weiss/ejabberd/blob/621f0e2b7cbb6c431e2c26ad6561c6c0eb74c065/src/ejabberd_c2s.erl#L3097